k8s/pkg/schedulers/scheduler_on_kubernetes.go

```diff
@@ -0,0 +1,185 @@
+/** Copyright 2020-2023 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package schedulers implements the vineyard scheduler plugin.
+package schedulers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/v6d-io/v6d/k8s/apis/k8s/v1alpha1"
+	"github.com/v6d-io/v6d/k8s/pkg/config/annotations"
+	"github.com/v6d-io/v6d/k8s/pkg/config/labels"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Scheduler is the interface for all vineyard schedulers
+type Scheduler interface {
+	Schedule(interface{}) (interface{}, error)
+}
+
+// SchedulerConfig is the common configuration for all schedulers
+type SchedulerConfig struct {
+	Required       []string
+	Nodes          []string
+	Namespace      string
+	OwnerReference *[]metav1.OwnerReference
+}
+
+// VineyardSchedulerOnKubernetes is the vineyard scheduler on kubernetes
+type VineyardSchedulerOnKubernetes struct {
+	client.Client
+	pod     *v1.Pod
+	rank    int
+	replica int
+	config  SchedulerConfig
+}
+
+// NewVineyardSchedulerOnKubernetes returns a new vineyard scheduler on kubernetes
+func NewVineyardSchedulerOnKubernetes(
+	c client.Client,
+	pod *v1.Pod,
+	rank int,
+	replica int,
+) *VineyardSchedulerOnKubernetes {
+	return &VineyardSchedulerOnKubernetes{
+		Client:  c,
+		pod:     pod,
+		rank:    rank,
+		replica: replica,
+	}
+}
+
+// SetupConfig setups the scheduler config
+func (vsok *VineyardSchedulerOnKubernetes) SetupConfig() error {
```

Comment:
`vsok` -> `vs` as there's no `vsoc` in `k8s/pkg/schedulers/scheduler_on_kubernetes.go` so would be no confusion.

Reply:
Make sense to me.